plugins/chat/app/services/chat/update_user_channel_membership.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Chat
+  # Service responsible for updating a user's channel membership settings.
+  # Currently supports updating the pinned status of a channel.
+  #
+  # @example
+  #  ::Chat::UpdateUserChannelMembership.call(
+  #    guardian: guardian,
+  #    params: {
+  #      channel_id: 1,
+  #      pinned: true
+  #    }
+  #  )
+  #
+  class UpdateUserChannelMembership
```

Comment:
I’m just wondering, do we want to create a service that will handle all the updating logic for memberships? Or should we create a very simple business action that would simply toggle the pinned status?